src/parser/mod.rs

```diff
@@ -1345,6 +1354,45 @@ impl<'a> Parser<'a> {
         })
     }
 
+    fn try_parse_method(&mut self, expr: &Expr) -> Result<Option<Expr>, ParserError> {
```

Comment:
instead of passing a reference we can use owned values, such that here we return the same expression if we didn't parse anything. That would avoid cloning the expression. I think with that and the linear representation of the method expression, the parsing logic could be slightly simpler (since it won't need to mutate anything, given that the maybe_parse section would only deal with the suffix)